### DISCUSSION TITLE
Can't render quarto doc with specific packages in vscode, but in RStudio

### DISCUSSION THREAD

**Discussion by werkstattcodes** [quarto-dev/quarto-cli#1372] [Q&A] [answered]
Hello,

I am encountering a strange behavior when using Quarto in vscode. When my script includes the packages writexl, summarytools or treemapify, I can't render the document. I can run the chunks without problem, but i can't render the document to html.  vscode/quarto tells me that the package doesn't exist.

Oddly, when I open the same file in RStudio, I can render the document without problem.

I already filed an issue in quarto/vscode rep, but maybe someone here encountered the issue as well and has already a remedy.

Many thanks.

---
title: "test_vscode_writexl"
format: html
---

```{r}
library(writexl)
library(summarytools)
library(treemapify)
```

Error in library(writexl) : there is no package called 'writexl'
Calls: .main ... withCallingHandlers -> withVisible -> eval -> eval -> library

**Comment by cderv** [comment, marked as answer]
When you are in vscode, 

* does this happen also with R Markdown document the same issue ? 
* if you open a R terminal in VS code, can you load the `writexl` package ? 
* Can you compare `.libPaths()` result from R console in R Studio and R terminal in VScode ? 
* Can you share results of `quarto check` from VSCODE terminal ?

**Reply by werkstattcodes** [reply]
Many thanks for your reply. Below my answers.

_does this happen also with R Markdown document the same issue ?_

I am encountering an issue when trying to render rmarkdown docs in vscode.

in vscode
> library(rmarkdown)
> pandoc_available()
[1] FALSE

in RStudio
```
> library(rmarkdown)
> pandoc_version()
[1] ‘2.18’
```

_if you open a R terminal in VS code, can you load the writexl package ?_
Yes

_Can you compare .libPaths() result from R console in R Studio and R terminal in VScode ?_
VScode 
`C:/Program Files/R/R-4.2.0/library"`

RStudio
```
[1] "C:/Users/Roland/AppData/Local/R/win-library/4.2"
[2] "C:/Program Files/R/R-4.2.0/library" 
```

_Can you share results of quarto check from VSCODE terminal ?_
Sorry, think i don't understand. What do you mean with "quarto check".

It seems there might be an issue with my .libPaths. But it's strange that i can call the packages interactively (comandline, chunks).

I added the "C:/Users/Roland/AppData/Local/R/win-library/4.2" under the vscode r extension setting R:lib paths. 

VSCode .libPaths() reads now as below, but the issue remains. I can't render the quarto document.
```
> .libPaths()
[1] "C:/Users/Roland/AppData/Local/R/win-library/4.2"
[2] "C:/Program Files/R/R-4.2.0/library"
```

**Reply by cderv** [reply]
Yes it seems the `.libPaths()` is not correctly loaded I wonder if quarto finds the same R version when rendering. I believe when running interactively, the R version set up in R extension in VSCODE is used. 

Are you using **rig** for R installation ? 

> Sorry, think i don't understand. What do you mean with "quarto check".

Open a terminal window in VS code and type `quarto check`

Also, you could try rendering the document outside for VSCODE in Terminal. It depends how you set up quarto also. 

`quarto check` should help us I believe.

**Reply by werkstattcodes** [reply]
Thanks again.
If I am reading the results correctly, you were right. Quarto is using R 4.0.3, not R 4.2. Any idea how I can change this?

```
[>] Checking Quarto installation......OK
      Version: 1.0.16
      Path: C:\Users\Roland\AppData\Local\Programs\Quarto\bin\
      CodePage: 1252

[>] Checking basic markdown render....OK

[>] Checking Python 3 installation....OK
      Version: 3.10.5
      Path: C:/Users/Roland/AppData/Local/Programs/Python/Python310/python.exe
      Jupyter: (None)

      Jupyter is not available in this Python installation.
      Install with py -m pip install jupyter

[>] Checking R installation...........OK
      Version: 4.0.3
      Path: C:/PROGRA~1/R/R-40~1.3
      LibPaths:
        - C:/Users/Roland/Documents/R/win-library/4.0
        - C:/Program Files/R/R-4.0.3/library
      rmarkdown: 2.11

[>] Checking Knitr engine render......OK
```

**Reply by cderv** [reply]
Do you have a R version set in your PATH ? Can you check that ? 
You can open variable environement windows on Windows  on in powershell window type 
````
$env:PATH -split ";"
````

It seems you have several R version installed, so you can also remove 4.0.3 if you don't use it anymore. That should solve the issue.

**Reply by werkstattcodes** [reply]
Deleted the older R versions. Issue solved. Many thanks (as so often)! This was very helpful.

**Comment by cderv** [comment]
For reference, issue related to this question: https://github.com/quarto-dev/quarto-vscode/issues/38
